### PR TITLE
misc: set predictable order for sample json timing entries

### PR DESCRIPTION
**Pull request by paulirish**
Everytime i run `yarn update:sample-json` i get a diff. even on master. It's the order of the timing entries.

I briefly looked into it. The behavior is different between node v15.2.0 and v16.10.0.  I'm guessing it had to do with [perf_hooks: web performance timeline compliance · nodejs/node@062f8e3](https://github.com/nodejs/node/commit/062f8e37307336d7f0a3e12f4ca252fe4b44a49d) .. as that landed in 16.6.0.

Anyhow. we'll just force a sort order so this won't happen. Luckily it means no diff for the json.

As a driveby i added a newline to the end of the file.. Yay for not having to see this anymore: 
image

### CODE REVIEW

File: lighthouse-core/scripts/cleanup-LHR-for-diff.js [hunk 1]
```diff
@@ -45,5 +46,6 @@ function cleanAndFormatLHR(lhrString) {
       auditResult.description = '**Excluded from diff**';
     }
   }
-  return JSON.stringify(lhr, null, 2);
+  // Ensure we have a final newline to confirm to .editorconfig
```
Comment (brendankenny) [inline lighthouse-core/scripts/cleanup-LHR-for-diff.js:49]:
```suggestion
  // Ensure we have a final newline to conform to .editorconfig
```

**Review by brendankenny — APPROVED** [review]
🕵️ 🎉

**Comment by Mbellsudteen** [comment]
ในวันที่ ส. 2 ต.ค. 2021 07:03 น. Brendan Kenny ***@***.***>
เขียนว่า:

**Comment by Mbellsudteen** [comment]
ในวันที่ ส. 2 ต.ค. 2021 06:54 น. Paul Irish ***@***.***>
เขียนว่า:

**Review by adamraine — APPROVED** [review]

File: lighthouse-core/scripts/cleanup-LHR-for-diff.js [hunk 2]
```diff
@@ -31,8 +31,9 @@ function cleanAndFormatLHR(lhrString) {
   // 2) CI gets a absolute path during yarn diff:sample-json
   lhr.configSettings.auditMode = true;
 
-  // Set timing values, which change from run to run, to predictable values
+  // Set timing values, which change from run to run, to predictable values and order
   lhr.timing.total = 12345.6789;
+  lhr.timing.entries.sort((a, b) => a.startTime - b.startTime);
```
Comment (connorjclark) [inline lighthouse-core/scripts/cleanup-LHR-for-diff.js:36]: why not sort in `runner.js` `_getTiming`?

Comment (paulirish) [reply lighthouse-core/scripts/cleanup-LHR-for-diff.js:36]: good idea. done.

**Pull request merged**